File: frontend/src/shared/types/InspirationImage.ts
import { InspirationType } from "./InspirationType";

/**
 * O poză din Galeria de Inspirație a proiectului: poză proprie, randare, sau inspirație online,
 * opțional legată de o cameră. `image`: URL http(s) sau `data:image/...;base64,...` (poză din telefon,
 * comprimată client-side) — aceeași convenție ca `Item.imageUrl`/`Offer.images`.
 */
export interface InspirationImage {
  id: string;
  projectId: string;
  roomId?: string;
  type: InspirationType;
  image: string;
  caption?: string;
  sourceUrl?: string;
  createdAt: string;
}

File: backend/src/main/java/ro/renovatorpro/domain/model/InspirationImage.java
package ro.renovatorpro.domain.model;

import java.time.Instant;
import java.util.Objects;

/**
 * O poză din Galeria de Inspirație a proiectului (CLAUDE.md backlog #4): poză proprie, randare, sau
 * inspirație preluată online, opțional legată de o cameră. {@code image} — URL http(s) sau
 * {@code data:image/...;base64,...} (aceeași convenție ca {@link Item#imageUrl()}/{@link Offer#images()}).
 *
 * <p>{@code roomId} rămâne {@code null} după ștergerea camerei ei (vezi V9 — {@code ON DELETE SET NULL}) —
 * pozele sunt conținut al userului, nu se pierd odată cu reconfigurarea apartamentului.
 */
public record InspirationImage(
        String id,
        String projectId,
        String roomId,
        InspirationType type,
        String image,
        String caption,
        String sourceUrl,
        Instant createdAt
) {

    public InspirationImage {
        Objects.requireNonNull(id, "id");
        Objects.requireNonNull(projectId, "projectId");
        Objects.requireNonNull(type, "type");
        Objects.requireNonNull(image, "image");
        Objects.requireNonNull(createdAt, "createdAt");
    }
}
